fix(tokens)_: enhance token collision solving logic, by giving preced…

vkjr opened the pull request:
Cherrypicking #6793 by @saledjenic into release branch. To be included in mobile patch

Giving precedence to the remote status list, we are able to add remotely token if it gets removed from the Uniswap list which we're using as a main reference. Remote status list is processed immediately after processing the Uniswap list.

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (8)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 46fb276c | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6794/1/) | 2025-08-01 10:19:42 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250801-101728-46fb27-pr6794.aar) |
| :heavy_check_mark: | 46fb276c | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6794/1/) | 2025-08-01 10:20:36 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250801-101728-46fb27-pr6794.zip) |
| :heavy_check_mark: | 46fb276c | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6794/1/) | 2025-08-01 10:21:50 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250801-101731-46fb27-pr6794.zip) |
| :x: | 46fb276c | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6794/1/) | 2025-08-01 10:22:00 | ~4 min | `windows` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6794/1/consoleText) |
| :heavy_check_mark: | 46fb276c | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6794/1/) | 2025-08-01 10:22:08 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250801-101733-46fb27-pr6794.zip) |
| :heavy_multiplication_x: | 46fb276c | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6794/1/) | 2025-08-01 10:22:50 | ~5 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6794/1//consoleText) |
| :heavy_check_mark: | 46fb276c | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6794/1/) | 2025-08-01 10:23:23 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250801-101736-46fb27-pr6794.zip) |
| :heavy_multiplication_x: | 46fb276c | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6794/1/) | 2025-08-01 10:35:32 | ~17 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6794/1//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | a232c2ff | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6794/2/) | 2025-08-04 09:51:41 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250804-094930-a232c2-pr6794.aar) |
| :x: | a232c2ff | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6794/2/) | 2025-08-04 09:52:24 | ~2 min | `windows` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6794/2/consoleText) |
| :heavy_check_mark: | a232c2ff | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6794/2/) | 2025-08-04 09:53:36 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250804-094935-a232c2-pr6794.zip) |
| :heavy_check_mark: | a232c2ff | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6794/2/) | 2025-08-04 09:54:59 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250804-094932-a232c2-pr6794.zip) |
| :heavy_check_mark: | a232c2ff | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6794/2/) | 2025-08-04 09:55:21 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250804-094940-a232c2-pr6794.zip) |
| :heavy_multiplication_x: | a232c2ff | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6794/2/) | 2025-08-04 09:57:44 | ~8 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6794/2//consoleText) |
| :heavy_check_mark: | a232c2ff | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6794/2/) | 2025-08-04 09:59:26 | ~9 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250804-094937-a232c2-pr6794.zip) |
| :heavy_check_mark: | a232c2ff | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6794/2/) | 2025-08-04 10:19:36 | ~29 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6794/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | d62e6337 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6794/3/) | 2025-08-06 16:15:17 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250806-161310-d62e63-pr6794.aar) |
| :heavy_check_mark: | d62e6337 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6794/3/) | 2025-08-06 16:16:17 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250806-161310-d62e63-pr6794.zip) |
| :x: | d62e6337 | [#3](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6794/3/) | 2025-08-06 16:17:26 | ~4 min | `windows` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6794/3/consoleText) |
| :heavy_check_mark: | d62e6337 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6794/3/) | 2025-08-06 16:17:37 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250806-161313-d62e63-pr6794.zip) |
| :heavy_check_mark: | d62e6337 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6794/3/) | 2025-08-06 16:17:49 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250806-161315-d62e63-pr6794.zip) |
| :heavy_check_mark: | d62e6337 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6794/3/) | 2025-08-06 16:18:57 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250806-161318-d62e63-pr6794.zip) |
| :heavy_multiplication_x: | d62e6337 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6794/3/) | 2025-08-06 16:20:26 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6794/3//consoleText) |
| :heavy_check_mark: | d62e6337 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6794/3/) | 2025-08-06 16:40:24 | ~27 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6794/3//consoleText) |

vkjr committed:
fix(tokens)_: enhance token collision solving logic, by giving precedence to the remote status list

Giving precedence to the remote status list, we are able to add remotely token if it gets removed from
the Uniswap list which we're using as a main reference. Remote status list is processed immediately
after processing the Uniswap list.

vkjr commented:
@igor-sirotin, could you please merge this one? The original #6793 has ben merged already
